File: 2015-determine-whether-matrix-can-be-obtained-by-rotation/determine-whether-matrix-can-be-obtained-by-rotation.py
def rotate(mat):
    n=len(mat)
    m=len(mat[0])
    res=[[0]*m for _ in range(n)]
    for i in range(n):
        for j in range(m):
            res[j][n-1-i]=mat[i][j]
    
    return res

class Solution:
    def findRotation(self, mat: List[List[int]], target: List[List[int]]) -> bool:
        val=mat
        for i in range(4):
            val=rotate(val)
            if val==target:
                return True
        
        return False
        

File: 2015-determine-whether-matrix-can-be-obtained-by-rotation/determine-whether-matrix-can-be-obtained-by-rotation.cpp
vector<vector<int>> rotate(vector<vector<int>>&mat){
    vector<vector<int>>res;
    int n=mat.size();
    int m=mat[0].size();
    for(int i=0;i<m;i++){
        vector<int>curr_col;
        for(int j=0;j<n;j++){
            curr_col.push_back(mat[j][i]);
        }
        reverse(curr_col.begin(),curr_col.end());
        res.push_back(curr_col);

    }
    return res;

}
class Solution {
public:
    bool findRotation(vector<vector<int>>& mat, vector<vector<int>>& target) {
        vector<vector<int>>temp=mat;
        for(int i=0;i<4;i++){
            
            if(temp==target)return true;
            temp=rotate(temp);
        }
        return false;
    }
};